base/setup/usetup/mui.c

```diff
@@ -535,6 +537,62 @@ SetConsoleCodePage(VOID)
     wCodePage = (UINT)wcstoul(MUIGetOEMCodePage(SelectedLanguageId), NULL, 10);
     SetConsoleOutputCP(wCodePage);
 #endif
+
+    SelectedCodePage = wCodePage;
+}
+
+CHAR
+MUITranslateChar(
+    IN CHAR Ch)
+{
+    switch (SelectedCodePage)
+    {
+        case 28606: /* Romanian */
+            switch ((BYTE)Ch)
+            {
+                case 0x07: Ch = 0x07; break; /* bullet */
+                case 0xDB: Ch = 0x01; break; /* block */
+                case 0xDD: Ch = 0x02; break; /* half-left block */
+                case 0x18: Ch = 0x03; break; /* up arrow */
+                case 0x19: Ch = 0x04; break; /* down arrow */
+                case 0xC4: Ch = 0x05; break; /* horizontal line */
+                case 0xB3: Ch = 0x06; break; /* vertical line */
+                case 0xDA: Ch = 0x08; break; /* upper left corner */
+                case 0xBF: Ch = 0x09; break; /* upper right corner */
+                case 0xC0: Ch = 0x0B; break; /* lower left corner */
+                case 0xD9: Ch = 0x0C; break; /* lower right corner */
+                case 0xC3: Ch = 0x0E; break; /* |- (vertical line and right horizontal line) */
+                case 0xB4: Ch = 0x0F; break; /* -| (left horizontal line and vertical line) */
+                case 0xCD: Ch = 0x10; break; /* double horizontal line (and underline) */
+                case 0xBA: Ch = 0x11; break; /* double vertical line */
+                case 0xC9: Ch = 0x12; break; /* double upper left corner */
+                case 0xBB: Ch = 0x13; break; /* double upper right corner */
+                case 0xC8: Ch = 0x14; break; /* double lower left corner */
+                case 0xBC: Ch = 0x15; break; /* double lower right corner */
+            }
+            break;
+
+        case 932: /* Japanese */
+            /* FIXME */
+            break;
+    }
+    return Ch;
+}
+
+VOID
+MUIUpdateVideoMode(VOID)
```

Comment:
This kind of functionality has to be moved in the console.c file, as part of the SetConsoleOutputCP() implementation.